src/main/java/org/wildfly/security/password/PasswordFactory.java

```diff
@@ -176,7 +176,7 @@ public Password translate(Password password) throws InvalidKeyException {
      * @return {@code true} if the guess matches the password, {@code false} otherwise
      * @throws InvalidKeyException if the given password is not supported by this factory
      */
-    public boolean verify(Password password, char[] guess) throws InvalidKeyException {
-        return spi.engineVerify(algorithm, password, guess);
+    public boolean verifyCredential(Password password, Object guess) throws InvalidKeyException {
```

Comment:
I don't think this API should be changed.  A password can only ever be character arrays (or strings, but this should generally be discouraged as per JCE best practice).  The minimal burden of conversion lies with the API consumer.